### PR TITLE
feat: split schematics localization key

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Proposed change
Split schematics to add localization key

### CODE REVIEW

File: packages/@o3r/localization/README.md [hunk 1]
```diff
@@ -714,7 +714,8 @@ it-IT **fallbacks to** en-GB, as it is the default fallback.
...
 
-| Schematics                 | Description                                                    | How to use                        |
-| -------------------------- | -------------------------------------------------------------- | --------------------------------- |
-| add                        | Include Otter localization module in a library / application.  | `ng add @o3r/localization`        |
-| localization-to-component  | Add localization to an Otter component                         | `ng g localization-to-component`  |
+| Schematics                     | Description                                                    | How to use                           |
+| ------------------------------ | -------------------------------------------------------------- | ------------------------------------ |
+| add                            | Include Otter localization module in a library / application.  | `ng add @o3r/localization`           |
+| localization-to-component      | Add localization to an Otter component                         | `ng g localization-to-component`     |
```
Comment (vscaiceanu-1a) [inline packages/@o3r/localization/README.md:720]:
```suggestion
| localization-to-component      | Add localization support to an Otter component                         | `ng g localization-to-component`     |
```

File: packages/@o3r/localization/schematics/add-localization-key/index.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+        translationsVariableType = member.type.typeName.escapedText.toString();
+        if (member.initializer && ts.isIdentifier(member.initializer)) {
+          defaultTranslationVariableName = member.initializer.escapedText.toString();
+        }
+      }
+    }
+  });
+
+  if (!localizationJsonPath || !tree.exists(localizationJsonPath)) {
+    throw new Error(
```
Comment (vscaiceanu-1a) [inline packages/@o3r/localization/schematics/add-localization-key/index.ts:51]: Actually, the coolest would be if this schematics will ask you if you want to run the `localization-to-component` (and run it for you)

File: packages/@o3r/localization/schematics/add-localization-key/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+      const runner = new SchematicTestRunner('schematics', collectionPath);
+
+      await expect(runner.runSchematic('add-localization-key', {
+        path: ngComponentPath,
+        key: 'dummyLoc1',
+        description: 'Dummy 1 description',
+        value: 'Dummy 1'
+      }, initialTree)).rejects.toThrow();
+
+      await expect(runner.runSchematic('add-localization-key', {
```
Comment (vscaiceanu-1a) [inline packages/@o3r/localization/schematics/add-localization-key/index.spec.ts:152]: I thin this expect should be in a dedicated test

Comment (matthieu-crouzet) [reply packages/@o3r/localization/schematics/add-localization-key/index.spec.ts:152]: as discussed will be done in a separate pr on all schematics

File: packages/@o3r/localization/schematics/add-localization-key/index.ts (hunk 2 above)
Comment (matthieu-crouzet) [reply packages/@o3r/localization/schematics/add-localization-key/index.ts:51]: as discussed will be done in another PR

Comment (matthieu-crouzet) [reply packages/@o3r/localization/schematics/add-localization-key/index.ts:51]: https://github.com/AmadeusITGroup/otter/issues/480

**Review by vscaiceanu-1a — APPROVED** [review]

File: packages/@o3r/localization/schematics/add-localization-key/index.ts [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+    ts.ScriptTarget.ES2015,
+    true
+  );
+
+  const o3rClass = componentSourceFile.statements.find((statement): statement is ts.ClassDeclaration => ts.isClassDeclaration(statement) && isO3rClassComponent(statement))!;
+  let localizationJsonPath: string | undefined;
+  let defaultTranslationVariableName: string | undefined;
+  let translationsVariableName: string | undefined;
+  let translationsVariableType: string | undefined;
+  o3rClass.members.forEach((member) => {
```
Comment (mrednic-1A) [inline packages/@o3r/localization/schematics/add-localization-key/index.ts:30]: What happens if o3rClass is undefined ? 
Should we throw saying that the component is not an otter component? 
Question aside: Is it mandatory to be an otter component ?

Comment (matthieu-crouzet) [reply packages/@o3r/localization/schematics/add-localization-key/index.ts:30]: `getO3rComponentInfo` is called before this method and will ensure we have an otter component

> Is it mandatory to be an otter component ?
That what we have done everywhere and if it is an angular component we propose to convert the component into an otter one before when https://github.com/AmadeusITGroup/otter/issues/480 will be done

**Review by mrednic-1A — APPROVED** [review]

**Review by vscaiceanu-1a — APPROVED** [review]

**Review by kpanot — APPROVED** [review]

**Pull request merged**